check.sh

```diff
@@ -20,10 +20,12 @@
 format_diff() {
 	local filepath="$1"
 	# Invoke clang-format with dry run and formatting error output
+	formatted="$(docker run -i -v "$(pwd)":"$(pwd)" -w "$(pwd)" --rm ghcr.io/jidicula/clang-format:"$CLANG_FORMAT_VERSION" --style=file --fallback-style="$FALLBACK_STYLE" "${filepath}")"
+	report_format="$(diff <(cat "${filepath}") <(echo "${formatted}"))"
+
```

Comment:
By moving this block here, we're going to be running clang-format twice on every file for versions >9, which will at least double the action's runtime - imo, this is an unacceptable loss in speed.

I like the idea of having a detailed report listing all the badly-formatted files, and uploading that as an artifact.

I don't think we should be including the output of the bad formatting into the report. It seems like useless output in a report, because once a developer gets a failing CI run in GitHub Actions when using this action, all they can do is to run the `clang-format` tool on the misformatted file and commit-push their changes.

I'm ok with having the bad format being output into the CI run itself, just to have it _somewhere_, but I don't think it belongs in a report. Ideally, I can see a report being used as the basis of a checklist for a developer to a) know which files they need to run clang-format on or b) know which paths to exclude from clang-format-action CI runs.

So overall, I'm requesting you to:
1. put the `formatted=` line back into the else-clause
2. remove the `report_format` declaration in this block
3. remove the `report_format` reference in the report creation step

Reply:
If you only want to add the list of the badly-formatted files, then please feel free to close this PR and do as you want in your version.
The reason I need this is that the result of the dry run is quite often useless. Most of the time it just reports that there is a problem, but gives no indication what is wrong. The `error: code should be clang-formatted [-Wclang-format-violations]` is the most useless piece of message I have ever read from an analytic tool...
I do not have or run clang-format locally only by github actions, this means I need a detailed report to understand what is wrong and how to fix it.